src/torchmetrics/classification/precision_recall_curve.py

```diff
@@ -41,6 +45,26 @@
 from torchmetrics.utilities import rank_zero_warn
 from torchmetrics.utilities.data import dim_zero_cat
 
+_DYNAMIC_THRESHOLDS_MIN_NSAMPLES = 1024**2  # 1MiB,TODO: find a better way to estimate a reasonable minimum
+
+
+def _budget_bytes_to_nsamples(budget_bytes: int):
+    # assume that both preds and target ("* 2") will be of size (N, 1) and of type float32 (4 bytes)
+    return budget_bytes / (2 * 4)
+
+
+def _validate_memory_budget(budget: int):
+    if budget <= 0:
+        raise ValueError("Budget must be larger than 0.")
+
+    if _budget_bytes_to_nsamples(budget) <= _DYNAMIC_THRESHOLDS_MIN_NSAMPLES:
+        warnings.warn(
+            f"Budget is relatively small ({humanfriendly.format_size(budget, binary=True)}). "
+            "The dynamic mode is recommended for bigger samples."
+        )
+
+    return budget
```

Comment:
if we only have the `thresholds` argument i guess all this logic can be moved to the `_adjust_threshold_arg` function

Reply:
In the multiclass/label cases the estimation (number of samples) <-> (memory consumption) would be different.

I don't have a very strong opinion on this, i will put some more thought on the multi* cases first.